Project1/Map.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <vector>
+#include "Unit.h"
+#include "GridCell.h"
+#include "GridCell.h"
+#include "CellPath.h"
+#include <map>
+#include <queue>
+/**
+Map class. Holds a Matrix of GridCells and all of the units. Should be initialized at the start of each chapter.
+*/
+using PathMap = std::map<GridCell*, CellCost>;
+using PathQueue = std::priority_queue<CellCost>;
+class Map {
+private:
+	std::map<Unit*, GridCell*> _unit_to_cell;
+	std::vector<std::vector<GridCell>> _grid;
+	std::vector<Unit> _units;
+	void insertAdjacencies();
+
+public:
+	Map(const Map& map) = delete;
+	Map& operator=(const Map & map) = delete;
+	Map(Map&& map) = default;
+	Map& operator=(Map&& map) = default;
+
+	Map(int width, int height);
+	
+	void moveUnit(GridCell& start, GridCell& destination);
+	void insertUnit(Unit& new_unit, GridCell& destination);
+	void removeUnit(Unit& unit);
+	
+	GridCell& getGridCell(int x_pos, int y_pos);
+	GridCell& getGridCell(Unit& unit);
+
+	const GridCell& getGridCell(int x_pos, int y_pos) const;
+	const GridCell& getGridCell(Unit& unit) const;
+
+	PathMap findShortestPaths(GridCell& start);
+	PathMap findShortestPaths(GridCell& start, int max_move, MobilityList<bool> mobility);
```

Comment:
Should add PathMap findShortestPaths(GridCell& start, int max_move, MobilityList<bool> mobility, bool intangible);